internal/gatewayapi/translator.go

```diff
@@ -101,12 +102,15 @@ type Translator struct {
 
 	// Namespace is the namespace that Envoy Gateway runs in.
 	Namespace string
+
+	Logger logging.Logger
```

Comment:
still seeing the `Logger` and `LoggedErrors` being defined here
can we follow the approach taken in the `gatewayapi` lib where we have a top level `errs` field e.g. https://github.com/envoyproxy/gateway/blob/cffaf27297d80349f78f7b56933880f7dae0cea7/internal/gatewayapi/backendtrafficpolicy.go#L299 , in this case it would be in the `Translate` function and we'd append the errors returned  from individual functions e.g. https://github.com/envoyproxy/gateway/blob/cffaf27297d80349f78f7b56933880f7dae0cea7/internal/gatewayapi/backendtrafficpolicy.go#L321

Reply:
The difference is that in `BackendTrafficPolicy` the errors are aggregated into a single `error` and then stored in the policy status. 

In this policy, the decision was not to update the policy status section for errors, which means that errors would need to be propagated all the way to the runner in order to be logged. This is a pretty large structural change.

I will remove the logger, but I think keeping the `LoggedErrors` approach is better.

Reply:
I was assuming this would be a best effort approach in the extension manager, but it looks like we've enhanced the gateway api translator to make sure the extension policy is conformant. In that case, lets populate the `Status` of the policy instead of using `LoggedErrors`

Reply:
When an extension policy doesn't implement the required contract - it has no `targetRef` or `targetRefs` attribute in its spec - then it's not possible to correctly place an error message in the policy status section. This is because finding the right place for the condition requires having the `targetRef`.

I think that there's value in logging this, though, to help with debugging sessions where the user wants to understand why the policy is not being sent to the extension server.